glommio/src/sys/uring.rs

```diff
@@ -1042,6 +1082,10 @@ impl UringCommon for SleepableRing {
                 _ => None,
             },
             |mut src, res| {
+                self.lat_req
+                    .entry(src.io_requirements.latency_req)
```

Comment:
I have a really bad feeling about this.
There are situations like linked sqe, cancellations, etc, that may create a mismatch here.
It is potentially safe but quite fragile, and at least we should have low level unit tests that test those cases specifically.

This is very tricky, because if this counter has a mismatch, it won't present itself as a bug, but as a very weird performance issue.

My idea of implementing this was to do this at a higher (executor) layer, and not at the uring layer. If requests are in-flight the task will be `Pending`, and we would include the presence of pending tasks when calculating the preempt timer.

Reply:
I agree. I felt the same when I wrote this. 

Doing this at the executor level means we would be including requests going through the latency ring. We would instead not influence the preemption timer since the latency ring already does that for us. 
But, because what goes in what ring is an implementation detail of the reactor, I felt like this stuff belonged there instead.

Now that this code, I like the idea of keeping two kinds of latency requirements: (1) the scheduler latency requirement and (2) the reactor latency requirement. I feel those two things are somewhat orthogonal. 

Your point is very much valid, and if we decide to go the route of this PR, this will require a rework.